remote_execution/oss/re_grpc/src/client.rs

```diff
@@ -312,6 +298,33 @@ impl REClientBuilder {
             return Err(anyhow::anyhow!("Server has remote execution disabled."));
         }
 
+        let max_decoding_msg_size = opts.max_decoding_message_size
+            .unwrap_or(capabilities.max_msg_size * 2);
+
+        if max_decoding_msg_size < capabilities.max_msg_size {
+            return Err(anyhow::anyhow!(
+                "Attribute `max_decoding_message_size` must always be equal or higher to `max_total_file_size`"));
+        }
+
+        let grpc_clients = GRPCClients {
+            cas_client: ContentAddressableStorageClient::with_interceptor(
+                cas.context("Error creating CAS client")?,
+                interceptor.dupe(),
+            ).max_decoding_message_size(max_decoding_msg_size),
+            execution_client: ExecutionClient::with_interceptor(
+                execution.context("Error creating Execution client")?,
+                interceptor.dupe(),
+            ).max_decoding_message_size(max_decoding_msg_size),
+            action_cache_client: ActionCacheClient::with_interceptor(
+                action_cache.context("Error creating ActionCache client")?,
+                interceptor.dupe(),
+            ).max_decoding_message_size(max_decoding_msg_size),
+            bytestream_client: ByteStreamClient::with_interceptor(
+                bytestream.context("Error creating Bytestream client")?,
+                interceptor.dupe(),
+            ).max_decoding_message_size(max_decoding_msg_size),
```

Comment:
It's not obvious to me that it's correct to apply a value that possibly computed specifically for the CAS client to all the other clients as well. Is there some justification for that?

I maybe just don't know enough about protobuf here as well. What's the consequence of not setting this value at all on the client?

Reply:
It wouldn't be a bad idea to have a separate value for each grpc client, I was just trying to keep things simple, but I wouldn't be against making this change. I am not sure users would normally tweak these values and aside from bytestream and cas methods all the others methods in the RE protocol tend to have generally small payloads so this limit never really kicks in, which is why I think it's generally safe to keep the same limit around for all of them and reduce the cognitive burden on the configuration. Wdyt?

Reply:
I don't know that I'm asking for different configuration for each client, I guess my suggestion is to just apply this value to the CAS client only and then leave the rest as they are. Especially if we think it's not necessary for those clients, that seems like the simplest model to me and prevents anyone from hitting issues where other clients change behavior as a result of modifying what looks like a CAS config

Reply:
Then how about we keep this value just for Bytestream and CAS clients? Wdyt?

Reply:
That sounds good